pennylane/tape/qnode.py

```diff
@@ -139,6 +140,11 @@ def __init__(
                 "Invalid device. Device must be a valid PennyLane device."
             )
 
+        if "shots" in inspect.signature(func).parameters:
+            raise qml.QuantumFunctionError(
+                "The shots argument is reserved and will be overwritten!"
+            )

```

Comment:
I would make this a warning rather than an exception. I can't quite explain, but it 'feels' more like a warning.

Like, we shouldn't stop the user from creating the qnode, but we should warn them that the shots argument won't act as they expect?

Reply:
The reason why I changed to the error is that I played around and all sorts of strange things happened until I realised that I used a shots argument in my code and I had just overlooked the warning... 

I think a warning is more appropriate but an exception safer? Happy to change though!